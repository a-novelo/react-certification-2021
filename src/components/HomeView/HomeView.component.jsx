import React from 'react';
import { Wrapper, HomeViewTitle, HomeViewCardContainer } from './HomeView.styled';
import Card from '../Card/Card.component';
import mockData from '../../resources/youtube-videos-mock.json';

function HomeView() {
  return (
    <Wrapper>
      <HomeViewTitle>Welcome to the Challenge!</HomeViewTitle>
      <HomeViewCardContainer>
        {mockData.items.map((video) => (
          <div key={video.etag}>
            <Card
              photoHeader={video.snippet.thumbnails.high.url}
              title={video.snippet.title}
              description={video.snippet.description}
            />
          </div>
        ))}
      </HomeViewCardContainer>
    </Wrapper>
  );
}

export default HomeView;
