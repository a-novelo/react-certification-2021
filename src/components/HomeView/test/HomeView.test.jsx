import React from 'react';
import renderer from 'react-test-renderer';
import HomeView from '../HomeView.component';

describe('#HomeView', () => {
  it('displays correctly', () => {
    const tree = renderer.create(<HomeView />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
