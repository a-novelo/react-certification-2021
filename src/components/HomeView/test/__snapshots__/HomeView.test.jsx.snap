// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#HomeView displays correctly 1`] = `
<div
  className="sc-bdfBwQ dMyNWP"
>
  <h2
    className="sc-gsTCUz cqjpmL"
  >
    Welcome to the Challenge!
  </h2>
  <div
    className="sc-dlfnbm jZPfxl"
  />
</div>
`;
